purchase_line_reassign/wizards/purchase_line_reassign_wiz.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, exceptions, fields, models
+
+
+class PurchaseOrderLineReassignWiz(models.TransientModel):
+    _name = "purchase.order.line.reassign.wiz"
+    _description = "Wizard to reassign purchase order lines to other PO"
+
+    def _default_partner_id(self):
+        active_ids = self.env.context.get("active_ids")
+        lines = self.env["purchase.order.line"].browse(active_ids)
+        partner = lines[:1].order_id.partner_id
+        if lines.filtered(lambda l: (l.invoice_lines or l.partner_id != partner)):
+            raise exceptions.ValidationError(
+                _("Selected line/s are invoiced or they have distinct vendors")
+            )
+        return partner
+
+    purchase_order_id = fields.Many2one(
+        comodel_name="purchase.order",
+        string="Move to purchase",
+    )
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Vendor",
+        default=lambda self: self._default_partner_id(),
+    )
+
+    def action_apply(self):
+        active_ids = self.env.context.get("active_ids")
+        lines = self.env["purchase.order.line"].browse(active_ids)
```

Comment:
Why browse active ids twice, instead of adding a line_ids field on the wizard?

Reply:
If I add a field means than I have do create operations in database. I do not need this due to performance